app/app-controllers/segmentedController.js

```diff
@@ -0,0 +1,78 @@
+define([
+    'bluebird',
+    'plugins/anchoredLayoutPlugin',
+    'plugins/segmentedPlugin',
+    'config/baseConfig',
+    'config/segmentsConfig'
+],
+function(
+    Promise,
+    AnchoredLayoutPlugin,
+    SegmentedPlugin,
+    BaseConfig,
+    SegmentsConfig
+) {
+    var SegmentedController = function(layout, navigationView, config, urlMap) {
+        this.parentLayout = layout;
+        this.navigationView = navigationView;
+        this.config = config;
+        this.urlMap = urlMap;
+        this.currentSegments = null;
+        this.registerEvents();
+    };
+
+    SegmentedController.init = function(layoutPromise, navigationViewPromise) {
+        var urlMap = {};
+        return Promise.join(
+            layoutPromise,
+            navigationViewPromise,
+            function(layout, navigationView) {
+                return Promise.all(
+                    SegmentsConfig.map(function(page, _) {
+                        return SegmentedPlugin.init().then(function(segmentedControls) {
+                            segmentedControls.setColor(BaseConfig.colors.primaryColor);
+                            segmentedControls.setItems(page.items);
+                            layout.addTopView(segmentedControls, {height: 44});
+                            layout.hideView(segmentedControls, {animated: false});
+                            urlMap[page.url] = segmentedControls;
+                        });
+                    })
+                ).then(function() {
+                    return new SegmentedController(layout, navigationView, SegmentsConfig, urlMap);
+                });
+            }
+        );
+    };
+
+    SegmentedController.prototype.loadSegments = function(url) {
```

Comment:
I was confused by the naming here. This ends up showing the segments for a given url. What do you think about `showSegmentsForUrl`?